Firestore/Source/Local/FSTLocalDocumentsView.mm

```diff
@@ -72,7 +75,7 @@ - (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key {
 // Internal version of documentForKey: which allows reusing `batches`.
 - (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key
                                     inBatches:(NSArray<FSTMutationBatch *> *)batches {
-  FSTMaybeDocument *_Nullable document = [self.remoteDocumentCache entryForKey:key];
+  FSTMaybeDocument *_Nullable document = _remoteDocumentCache->Get(key);
```

Comment:
A few thoughts on this API:

  * We have `Get` but `AddEntry`
  * We have `AddEntry` but `GetMatchingDocuments`

The nouns (or the lack thereof) don't really match.